x-pack/legacy/plugins/lens/public/indexpattern_datasource/indexpattern_suggestions.ts

```diff
@@ -197,16 +197,29 @@ function addFieldAsMetricOperation(
     field,
   });
   const newColumnId = generateId();
-  const updatedColumns = {
-    ...layer.columns,
-    [newColumnId]: newColumn,
-  };
-  const updatedColumnOrder = [...layer.columnOrder, newColumnId];
+
+  const [, metrics] = separateBucketColumns(layer);
+
+  // Add new metric if we already have multiple
+  if (metrics.length > 1) {
```

Comment:
We currently not using this path if `metrics.length === 0`, and then later executing code like `delete newColumns[metrics[0]]` where `metrics[0]` is `undefined`. This happens to work, but I think it would be a bit nicer, if we would also use this path if `metrics.length === 0`, since we don't then access `metrics[0]` and it could look a bit more clear. (At that point maybe we want to switch that if with the code below and make it a `metrics.length === 1`-if instead.